src/datastore/postgres-store.ts

```diff
@@ -3754,6 +3987,34 @@ export class PgDataStore
 
   cachedParameterizedInsertStrings = new Map<string, string>();
 
+  generateParameterizedWhereAndOrClause(
+    column1: string,
+    column2: string,
+    valuesCount: number
+  ): string {
+    const condition =
+      '(' +
+      column2 +
+      '=$' +
+      valuesCount * 2 +
+      ' AND ' +
+      column1 +
+      '=$' +
+      (valuesCount * 2 - 1) +
+      ')'; // e.g. (tx_id=$1 AND index_block_hash)
+    if (valuesCount === 0) {
+      return '';
+    } else if (valuesCount === 1) {
+      return condition;
+    }
+
+    return (
+      condition +
+      ' OR ' +
+      this.generateParameterizedWhereAndOrClause(column1, column2, valuesCount - 1)
+    );
```

Comment:
I'm still wary of the query string generation here. We've previously only done something like this for batch inserts of large datasets (where dynamically generating parameterized batch-insert queries is one of the fastest methods).

If generating a query string for this is absolutely necessary, we should increase the readability of this code. For example, it could use the syntax `WHERE (a,b) = ANY(VALUES (1,1),(1,2))` to avoid generating a string with a bunch of conditionals and redundant column names. Also, if this code is only called by one function, it would be more clear if it was defined within that function.

Reply:
Done